scripts/getting_started.sh

```diff
@@ -0,0 +1,583 @@
+     ## check  if denpency is installed before
+    isCmdInstalled "${cmdName}"
+
+    ## install denpency if it not installed
+    if [ $? -eq 0 ] ; then
+        echo "Installing ${cmdName}..."
+        curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.0.0/clusterctl-linux-amd64 -o clusterctl && sudo install clusterctl /usr/local/bin/clusterctl
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+
+function intallDependencies(){
+    installDocker
+    enableDocker
+    installKind
+    installClusterctl
+    commonInstall jq "sudo apt install -y jq"
+    commonInstall kubectl "sudo snap install kubectl --classic"
+    commonInstall make "sudo apt install -y build-essential"
+    commonInstall go "sudo snap install go --classic"
+}
+
+function commonInstall(){
+    cmdName=$1
+    installCmd=$2
+
+    ## check  if denpency is installed before
+    isCmdInstalled "${cmdName}"
+
+    ## install denpency if it not installed
+    if [ $? -eq 0 ] ; then
+        runCmd "${installCmd}" 0 "Installing ${cmdName}..."
+
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+function createKindCluster(){
+    echo  "Creating kind cluster..."
+
+    msg=$(kind create cluster 2>&1)
+    if [ $? -ne 0 ]; then
+        echo "Create kind cluster failed"
+        echo $msg | grep -q "You have reached your pull rate limit"
+        if [ $? -eq 0 ]; then 
+            echo "Suggestion: you can use \"docker login\" to avoid such an error."
+            exit
+        fi
+    else
+        echo "Create kind cluster successfully"
+    fi
+}
+
+function downloadByohCode(){
+    runCmd "rm -rf cluster-api-provider-bringyourownhost" 0 " Cleaning byoh code..."
+    msg=$(git clone git@github.com:vmware-tanzu/cluster-api-provider-bringyourownhost.git 2>&1)
+    if [ $? -ne 0 ]; then
+        echo "Downloading byoh code..."
+        echo $msg | grep "Please make sure you have the correct access rights"
+        if [ $? -eq 0 ]; then 
+            echo "Suggestion: Add an public key of this machine into \"SSH and GPG keys\" of your github setting"
+            exit
+        fi
+    else
+        echo "Download byoh code successfully"
+    fi
+}
+
+export PATH=/snap/bin:${PATH}
+intallDependencies 
+ 
+runCmd "sudo swapoff -a" 0 "Turning off swap..."
+swapMsg=$(sudo swapon -s)
+if [ -n "${swapMsg}" ]; then
+    echo "Please turn off swap first."
+    exit
+fi
+ 
+# check if cluster "kind" is already exited
+clusterName=`kind get clusters`
+if [ "${clusterName}" != "kind" ]; then
+    #runCmd "kind create cluster" 0 "Creating kind cluster..."
+    createKindCluster
+fi
+
+configClusterctl
+
+
+# check if init it before
+kubectl get pods --all-namespaces | grep -q byoh-controller-manager
+if [ $? -ne 0 ]; then
```

Comment:
Do we need to check this? What if we `init` even when provider is already installed? If it doesn't throw an error, we should be good, right?

Reply:
I found if we init again, it will report error and not continue this script, even caused some pod crash sometimes.
My script are designed that it can re-run again, again and again.